### PR TITLE
Improve contributing document

### PR DESCRIPTION
**Pull request by wolf99**
This  hopefully fleshes out the CONTRIBUTING.md document a little more, especially with regard to the point of view of someone for who this may be their first exposure to contributing to exercism, open source in general or even use of git.

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -8,12 +8,61 @@ Firstly, have a read of the Exercism [docs repository](https://github.com/exerci
...
+
+For higher-level discussions about the project as a whole check out the issues in the [discussions](https://github.com/exercism/discussions/issues) repository.
+
+### Good First Patch
+
+We're trying to label issues with "good first patch" if we think that these can be solved without too much context about exercism.io's codebase or functionality. To find them, you can do a [search](https://github.com/exercism/c/good-first-patch).
+
+## Pull Requests
+
+It would be awesome if you follow this [simple guidelines](https://github.com/exercism/docs/blob/master/contributing/pull-request-guidelines.md) while creating your PR, but it isn't required.
```
Comment (ryanplusplus) [inline docs/CONTRIBUTING.md]: `this` => `these`

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -8,12 +8,61 @@ Firstly, have a read of the Exercism [docs repository](https://github.com/exerci
...
+
+Pull requests should be focused on a single, issue or change.
+We welcome changes to code style and wording, please open a separate PR for these changes if possible.
```
Comment (ryanplusplus) [inline docs/CONTRIBUTING.md]: `please` => `but please`?

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -44,7 +120,7 @@ The good folks at fish-shell have [some great information](https://github.com/fi
 
 ### Run CI Scripts Locally
 You can also run individual scripts on your own machine before commiting. 
-Firstly make sure you have the necessary tools installed (such as `indent`, `git`, `sed`, `make`, `valgrind` and a C compiler), and then run the required script from the repoistory root. For example:
+Firstly make sure you have the necessary tools installed (such as `indent`, [`git`](https://git-scm.com/), [`sed`](https://www.gnu.org/software/sed/manual/sed.html), [`make`](https://www.gnu.org/software/make/), [`valgrind`](http://valgrind.org/) and a C compiler), and then run the required script from the repoistory root. For example:
```
Comment (ryanplusplus) [inline docs/CONTRIBUTING.md]: `repoistory` => `repository`

**Comment by ryanplusplus** [comment]
Just a few nitpicks. Thanks for all of this work; this is a huge improvement.

**Comment by wolf99** [comment]
Thanks for the review @ryanplusplus. the latest commit fixes those and a few other typos I found.

**Pull request merged**